src/Domain/FunctionsOfTime/Tags.cpp

```diff
@@ -3,31 +3,92 @@
 
 #include "Domain/FunctionsOfTime/Tags.hpp"
 
+#include <cstddef>
 #include <memory>
+#include <optional>
+#include <tuple>
 
 #include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
 #include "Domain/Domain.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/ReadSpecThirdOrderPiecewisePolynomial.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
 
-namespace domain {
-namespace Tags {
+namespace domain::Tags {
 template <size_t Dim>
-auto InitialFunctionsOfTime<Dim>::create_from_options(
+auto InitialFunctionsOfTime<Dim, true>::create_from_options(
+    const std::unique_ptr<::DomainCreator<Dim>>& domain_creator,
+    const std::optional<std::string>& function_of_time_file,
+    const std::optional<std::map<std::string, std::string>>&
+        function_of_time_name_map) noexcept -> type {
+  if (function_of_time_file and function_of_time_name_map) {
+    // Currently, only support order 3 piecewise polynomials.
+    // This could be generalized later, but the SpEC functions of time
+    // that we will read in with this action will always be 3rd-order
+    // piecewise polynomials
+    constexpr size_t max_deriv{3};
+    std::unordered_map<std::string,
+                       domain::FunctionsOfTime::PiecewisePolynomial<max_deriv>>
+        spec_functions_of_time{};
+    importers::read_spec_third_order_piecewise_polynomial(
+        make_not_null(&spec_functions_of_time), *function_of_time_file,
+        *function_of_time_name_map);
+
+    std::unordered_map<std::string,
+                       std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+        functions_of_time{domain_creator->functions_of_time()};
```

Comment:
I got in the habit of preferring brace initialization whenever possible, because it protects against unintended conversions (for numerical types, not necessarily applicable here). Is there a reason you prefer =?

I avoided auto because I thought our preference in spectre is to specify the types and only use auto when the type is specified elsewhere on the line.

I'm not fussy about this, but I would like to know what the consensus is for this question. @nilsdeppe, do you have an opinion?